src/_pytest/outcomes.py

```diff
@@ -57,10 +57,13 @@ def __init__(self, msg="unknown reason"):
 # exposed helper methods
 
 
-def exit(msg):
+def exit(msg, returncode=None):
     """ exit testing process as if KeyboardInterrupt was triggered. """
     __tracebackhide__ = True
-    raise Exit(msg)
+    if returncode:
+        raise SystemExit(returncode)
+    else:
+        raise Exit(msg)
```

Comment:
I think we should always raise `Exit`, but pass the argument along to it: `Exit(msg, returncode)`, and treat it accordingly in `main.py`:

https://github.com/pytest-dev/pytest/blob/7e1fac5f9197998e93a836b6ea4393429cb09147/src/_pytest/main.py#L185-L188

Like this:

```python
existstatus = EXIT_INTERRUPTED
if initstate < 2 and isinstance(excinfo.value, exit.Exception):
    sys.stderr.write("{}: {}\n".format(excinfo.typename, excinfo.value.msg))
    exitstatus = excinfo.value.exitstatus
config.hook.pytest_keyboard_interrupt(excinfo=excinfo)
session.exitstatus = exitstatus
```

Reply:
Also, please update the docstring for ``pytest.exit``